merkato/exchanges/exchange.py

```diff
@@ -0,0 +1,79 @@
+from merkato.exchanges.tux_exchange import TuxExchange
+
+DEBUG = True
+
+class Exchange(object):
+    '''Merkato Market Making Bot Exchange Interface
+        This class acts as an entry point for all exchange interfaces.
+        handling delegation to the proper exchange interface (tux, polo, etc), retry logic, etc.
+    '''
+    def __init__(self, configuration):
+        self.exchange   = configuration['exchange']
+        self.DEBUG = 100 # TODO: move to configuration object
+
+        self.interface = None
```

Comment:
1. Why did you choose to have the exchange class to be a base class as opposed to using multiple utility methods?
2. Why did you choose the Base Exchange class to interact with the exchange specific class instead of owning the methods itself?
 i.e. why did you make the decision to have the Exchange sell method call self.interface.sell instead of self._sell where the base Exchange class owns the self._sell method directly?

Reply:
1. exchanges _could_ be a collection of methods rather than a class, since they don't have much shared data except a URL and not (currently) any shared state. but that could change. And just generally I am used to this kind of object-oriented design and find it helpful especially as projects grow large.

2. We could do it the second way, where Exchange.sell calls self._sell, and TuxExchange derives Exchange and implements self._sell, self._buy etc. Then you don't need BaseExchange. However, since you do have multiple types of exchange that all should expose a common interface, it makes sense to have a base class, if nothing else as a kind of code-as-documentation that shows what all methods an exchange needs to implement. Which may make it clearer for people implementing new exchanges in the future.
Also, this makes it pretty easy to put the responsibility of switching on configuration["exchange"] and creating the appropriate one here -- so the caller can just create an Exchange() and pass the appropriate config -- all it has to do is set self.interface(). If we do it the other way, the next level up (the Merkato class, currently) is responsible for instantiating the appropriate subclass. Which like, it could do, but ideally it knows as little about the exchange layer as possible, and can just issue commands without having to worry.